Evaluator.java

```diff
@@ -1,31 +1,134 @@
-public class HeuristicNew {
-	public static final int NUM_FEATURES = 6;
-	public static final boolean db = false; // set true to turn debug statements on
-    public static final boolean db2 = false; // set true to debug rows eliminated (recommended set db = false)
-	double[] weights;
-	int weightCounter;
-    TempState s;
-
-    public HeuristicNew(TempState s, double[] inputWeights) {
-        this.weights = inputWeights;
-        this.s = s;
-        this.weightCounter = 0;
-    }
+import java.util.Arrays;
+public class Evaluator {
+	private static final int TOTAL_NUM_FEATURES = 9;
+
+	// Indices for each feature to keep track of which feature to use
+	private static final int INDEX_ROWS_CLEARED = Heuristic.INDEX_ROWS_CLEARED;
+	private static final int INDEX_COL_HEIGHT = Heuristic.INDEX_COL_HEIGHT;
+	private static final int INDEX_COL_HEIGHT_DIFF = Heuristic.INDEX_COL_HEIGHT_DIFF;
+	private static final int INDEX_MAX_COL_HEIGHT = Heuristic.INDEX_MAX_COL_HEIGHT;
+	private static final int INDEX_NUM_HOLES = Heuristic.INDEX_NUM_HOLES;
+	private static final int INDEX_LANDING_HEIGHT = Heuristic.INDEX_LANDING_HEIGHT;
+	private static final int INDEX_ROW_TRANSITIONS = Heuristic.INDEX_ROW_TRANSITIONS;
+	private static final int INDEX_COL_TRANSITIONS = Heuristic.INDEX_COL_TRANSITIONS;
+	private static final int INDEX_WELL_SUMS = Heuristic.INDEX_WELL_SUMS;
+
+	// Flags for features
+	private char[] featureFlags;
+	private int numFeatures;
+	private double[] weights;
+	private int weightCounter;
+	private TempState s;
+	private int[] colHeights;
+
+	public Evaluator(Heuristic h, TempState s, double[] inputWeights) {
+		this.featureFlags = h.getFeatureFlags();
+		this.numFeatures = h.getNumFeatures();
+		this.weightCounter = 0;
+		this.weights = inputWeights;
+		this.s = s;
+	}
 
 	public double evaluate() {
 		int[][] field = s.getField();
-        if (db) s.printField(s.getField());
+		this.colHeights = colHeights(field);
+
+		return weightedNumRowsCleared(s.getRowsCleared()) 
+				+ weightedSumColHeight(colHeights) 
+				+ weightedSumColDiff(colHeights) 
+				+ weightedMaxColHeight(colHeights) 
+				+ weightedNumHoles(field, colHeights) 
+				+ weightedLandingHeight(s) 
+				+ weightedNumRowTranstions(field) 
+				+ weightedNumColTranstions(field)
+                + weightedWellSums(field);
+	}
+
+
+	/** METHODS TO COMPUTE FEATURE SUM **/
+	// Returns weighted number of rows cleared by playing this move
+	private double weightedNumRowsCleared(int rowsCleared) {
```

Comment:
Parameter `rowsCleared` is never used. Since state s is now class attribute of Evaluator, the parameters for this method can be empty.

Reply:
Good catch!